pep/app/pods/application/route.ts

```diff
@@ -63,12 +64,23 @@ export default class Application extends PageLayout(Route.extend(ApplicationRout
         // delay (and as a result, FOUC) to the app setup process
         // since we do not currently cache ajax service requests
         // in the fastboot shoebox (like ember-data does)
-        if (!this.session.isAuthenticated && this.fastboot.isFastBoot) {
-            try {
-                await this.session.authenticate('authenticator:ip');
-            } catch (errors) {
-                // fail silently - we always want to just try and get authenticated by IP and if it
-                // doesn't work thats fine
+
+        if (this.fastboot.isFastBoot) {
+            // If your not logged in yet - give IP auth a try
+            if (!this.session.isAuthenticated) {
+                try {
+                    await this.session.authenticate('authenticator:ip');
+                } catch (errors) {
+                    this.session.invalidate();
+                }
+                // If you are logged in, but your currently authenticated using IP - try again to ensure you didn't change locations
+                // But here we don't wait for the promise to return as that would cause a FOUC
+            } else if (this.session?.data?.authenticated?.SessionType === SessionType.IP) {
+                try {
+                    this.session.authenticate('authenticator:ip');
```

Comment:
how would `await`ing this cause a FOUC if its only being run in fastboot?

Reply:
Im not 100% sure, but it does. I believe the FOUC actually happens in FastBoot.

Reply:
does the FOUC not happen in the above case though, where if you are not logged in, we do `await` the authenticate call? I had seen something similar on stage recently though that sounds similar, where the site will appear to render w/o the stylesheet loaded, not sure if its related, but maybe.

Also, if we can't `await` here, we should remove the try/catch around it, as since you are not awaiting the response, the `catch` will never be hit if it fails. 

(also this seems like it could result in a race condition possibly, since its not blocking the fastboot request from returning, the response could be sent to the client before this has a chance to set cookies/etc, seems a bit weird.)

Reply:
Correct. So if you are not logged in and try the IP auth there is no FOUC, if you are logged in and try the IP auth it does. Which is why I put it as a separate block

Reply:
yea that part is fine for now (though I think we should figure out the real issue of why FOUC happens in one case but not the other when we get some time).

but i think we should at least remove the try/catch on the one that doesnt have the `await`, because its not doing anything. since you are not waiting for a response, a failed authenticate request will never hit that `catch`.

Reply:
That second else if is checking if you are still logged in correctly. So if your logged into IP already, we try to re auth you. In 99% of cases, there will be no change and in the 1% that there is, we should just log you out when that comes back